compiler/src/test/scala/org/scalajs/nscplugin/test/JSInteropTest.scala

```diff
@@ -1856,200 +1861,30 @@ class JSInteropTest extends DirectTest with TestHelpers {
-    }
-    """ hasErrors
-    """
-      |newSource1.scala:7: error: Native JS members inside non-native objects must have an explicit name in @JSGlobal
-      |      @JSGlobal
-      |       ^
-    """
-
-    """
-    object A {
-      @js.native
-      @JSGlobal
-      object B extends js.Object
-    }
-    """ hasErrors
-    """
-      |newSource1.scala:7: error: Native JS members inside non-native objects must have an explicit name in @JSGlobal
-      |      @JSGlobal
-      |       ^
-    """
 
-    // From issue #2401
-    """
-    package object A {
+    object Container {
       @js.native
       object B extends js.Object
-
-      @js.native
-      @JSGlobal
-      object C extends js.Object
     }
     """ hasErrors
     """
-      |newSource1.scala:7: error: Native JS objects must have exactly one annotation among @JSGlobal, @JSImport and @JSGlobalScope.
+      |newSource1.scala:6: error: Native JS objects must have exactly one annotation among @JSGlobal, @JSImport and @JSGlobalScope.
+      |    object A extends js.Object
+      |           ^
+      |newSource1.scala:10: error: Native JS objects must have exactly one annotation among @JSGlobal, @JSImport and @JSGlobalScope.
       |      object B extends js.Object
       |             ^
     """
-
-    """
-    package object A {
-      @js.native
-      class B extends js.Object
-
-      @js.native
-      @JSGlobal
-      class C extends js.Object
-    }
-    """ hasErrors
-    """
-      |newSource1.scala:7: error: Native JS classes, vals and defs must have exactly one annotation among @JSGlobal and @JSImport.
-      |      class B extends js.Object
-      |            ^
-    """
-
-    """
-    object A {
-      @JSName("InnerB")
-      @js.native
-      class B extends js.Object
-
-      @JSName("InnerC")
-      @js.native
-      abstract class C extends js.Object
-
-      @JSName("InnerD")
-      @js.native
-      object D extends js.Object
-    }
-    """ hasErrors
-    """
-      |newSource1.scala:6: error: @JSName can only be used on members of JS types.
-      |      @JSName("InnerB")
-      |       ^
-      |newSource1.scala:8: error: Native JS classes, vals and defs must have exactly one annotation among @JSGlobal and @JSImport.
-      |      class B extends js.Object
-      |            ^
-      |newSource1.scala:10: error: @JSName can only be used on members of JS types.
-      |      @JSName("InnerC")
-      |       ^
-      |newSource1.scala:12: error: Native JS classes, vals and defs must have exactly one annotation among @JSGlobal and @JSImport.
-      |      abstract class C extends js.Object
-      |                     ^
-      |newSource1.scala:14: error: @JSName can only be used on members of JS types.
-      |      @JSName("InnerD")
-      |       ^
-      |newSource1.scala:16: error: Native JS objects must have exactly one annotation among @JSGlobal, @JSImport and @JSGlobalScope.
-      |      object D extends js.Object
-      |             ^
-    """
```

Comment:
Is there a reason this test case got removed?

Reply:
This test, and the others pointed out below, seemed to duplicate what is also tested by:
* `noJSNameAnnotOn{Class,Object,Trait}`
* `native{Class,Object}HasLoadingSpec`

I think they were there because, historically, the differences between `@JSName` and `@JSGlobal` were a bit blurry. There was a period during 0.6.x (or was it 0.5.x) when `@JSGlobal` was implicit, and `@JSName` could be used to provide a name. These possibilities are long gone, and so the variants of the test here don't seem to carry their weight anymore, compared to the existing tests I mentioned above.